x-pack/plugins/ml/public/application/jobs/new_job/common/job_validator/validators.ts

```diff
@@ -39,39 +63,105 @@ export function isCardinalityModelPlotHigh(
 export function cardinalityValidator(
   jobCreator$: Subject<JobCreator>
 ): Observable<CardinalityValidatorResult> {
+  return combineLatest([
+    jobCreator$.pipe(pluck('modelPlot')),
+    jobCreator$.pipe(
+      filter((jobCreator) => {
+        return jobCreator?.modelPlot;
+      }),
+      map((jobCreator) => {
+        return {
+          jobCreator,
+          analysisConfigString: JSON.stringify(jobCreator.jobConfig.analysis_config, null, 2),
+        };
+      }),
+      distinctUntilChanged((prev, curr) => {
+        return prev.analysisConfigString === curr.analysisConfigString;
+      }),
+      switchMap(({ jobCreator }) => {
+        // Perform a cardinality check only with enabled model plot.
+        return ml
+          .validateCardinality$({
+            ...jobCreator.jobConfig,
+            datafeed_config: jobCreator.datafeedConfig,
+          } as CombinedJob)
+          .pipe(
+            map((validationResults) => {
+              for (const validationResult of validationResults) {
+                if (isCardinalityModelPlotHigh(validationResult)) {
+                  return {
+                    highCardinality: {
+                      value: validationResult.modelPlotCardinality,
+                      severity: VALIDATOR_SEVERITY.WARNING,
+                    },
+                  };
+                }
+              }
+              return null;
+            })
+          );
+      }),
+      startWith(null)
+    ),
+  ]).pipe(
+    map(([isModelPlotEnabled, cardinalityValidationResult]) => {
+      return isModelPlotEnabled ? cardinalityValidationResult : null;
+    })
+  );
+}
+
+export function jobIdValidator(jobCreator$: Subject<JobCreator>): Observable<JobExistsResult> {
   return jobCreator$.pipe(
-    // Perform a cardinality check only with enabled model plot.
-    filter((jobCreator) => {
-      return jobCreator?.modelPlot;
+    map((jobCreator) => {
+      return {
+        jobId: jobCreator.jobId,
+      };
+    }),
+    // No need to perform an API call if the analysis configuration hasn't been changed
+    distinctUntilChanged((prev, curr) => {
+      return prev.jobId === curr.jobId;
+    }),
+    switchMap(({ jobId }) => {
+      return ml.jobs.jobsExist$([jobId], true);
```

Comment:
nit: just noticed you can simplte return a `jobId` string in `map`. same with the `groups` check 
```suggestion
    map((jobCreator) => jobCreator.jobId),
    // No need to perform an API call if the analysis configuration hasn't been changed
    distinctUntilChanged((prevJobId, currJobId) => prevJobId === currJobId),
    switchMap(jobId => {
      return ml.jobs.jobsExist$([jobId], true);
```